src/analysis/inner_product_space/adjoint.lean

```diff
@@ -186,6 +186,37 @@ lemma is_adjoint_pair (A : E' →L[ℝ] F') :
 
 end real
 
+lemma norm_map_of_unitary {U : (E →L[𝕜] E)} (hU : U ∈ unitary (E →L[𝕜] E)) (x : E) :
+  ∥U x∥ = ∥x∥ :=
+begin
+  rw [← sq_eq_sq (norm_nonneg (U x)) (norm_nonneg x), norm_sq_eq_inner,
+    ← continuous_linear_map.adjoint_inner_left, ← continuous_linear_map.mul_apply],
+  rw unitary.mem_iff at hU,
+  rw [← continuous_linear_map.star_eq_adjoint, hU.1, continuous_linear_map.one_apply,
+    inner_self_eq_norm_sq],
+end
+
+lemma op_norm_of_unitary [nontrivial E] {U : (E →L[𝕜] E)} (hU : U ∈ unitary (E →L[𝕜] E)) :
+  ∥U∥ = 1 :=
```

Comment:
This follows directly from `cstar_ring.norm_of_mem_unitary` -- the only thing missing is an instance showing that `[nontrivial E]` implied `[nontrivial (E →L[𝕜] E)]` (which would be nice to add!).

Reply:
I see, I probably would not have thrown this in had I seen that.  It sounds like there's still something to do, though.  I don't know how to insert an instance that shows this; I have only worked with local instances, I think.

Reply:
Basically you would want to prove something like:
```
instance [nontrivial E] : nontrivial (E →L[𝕜] E) := sorry
```
(You might also need `[nontrivial 𝕜]`.) This should probably go close to where continuous linear maps are defined, so somewhere in `topology/algebra/module/basic.lean`.

Reply:
I don't think you should need `[nontrivial 𝕜]`. That should already be present for fields.

Reply:
I did not need `[nontrivial 𝕜]`.

I pushed the instance, which I think renders the lemma unnecessary.  Let me know if I have misunderstood.